modules/machine-user-infra-machines-ibm-z.adoc

```diff
@@ -0,0 +1,148 @@
+
+.Prerequisites
+
+* A domain name server (DNS) that can perform hostname and reverse lookup for the nodes.
+* An HTTP or HTTPS server running on your provisioning machine that is accessible to the machines you create.
+
+.Procedure
+
+. Extract the Ignition config file from the cluster by running the following command:
++
+[source,terminal]
+----
+$ oc extract -n openshift-machine-api secret/worker-user-data-managed --keys=userData --to=- > worker.ign
+----
+
+. Upload the `worker.ign` Ignition config file you exported from your cluster to your HTTP server. Note the URL of this file.
+
+. You can validate that the Ignition file is available on the URLs. The following example gets the Ignition config file for the compute node:
++
+[source,terminal]
+----
+$ curl -k http://<HTTP_server>/worker.ign
+----
+
+. Download the {op-system-base} live kernel, initramfs, and rootfs files by running to following commands:
++
+[source,terminal]
+----
+curl -LO $(oc -n openshift-machine-config-operator get configmap/coreos-bootimages -o jsonpath='{.data.stream}' \
+| jq -r '.architectures.s390x.artifacts.metal.formats.pxe.kernel.location')
+----
++
+[source,terminal]
+----
+curl -LO $(oc -n openshift-machine-config-operator get configmap/coreos-bootimages -o jsonpath='{.data.stream}' \
+| jq -r '.architectures.s390x.artifacts.metal.formats.pxe.initramfs.location')
+----
++
+[source,terminal]
+----
+curl -LO $(oc -n openshift-machine-config-operator get configmap/coreos-bootimages -o jsonpath='{.data.stream}' \
+| jq -r '.architectures.s390x.artifacts.metal.formats.pxe.rootfs.location')
+----
+
+. Move the downloaded {op-system-base} live kernel, initramfs, and rootfs files to an HTTP or HTTPS server that is accessible from the z/VM guest you want to add.
+
+. Create a parameter file for the z/VM guest. The following parameters are specific for the virtual machine:
+** Optional: To specify a static IP address, add an ip= parameter with the following entries, separated by colons (“:”):
+... The IP address for the machine.
+... An empty string.
+... The gateway.
+... The netmask.
+... The machine host and domain name in the form `hostname.domainname`. Omit this value to let {op-system} decide.
+... The network interface name. Omit this value to let {op-system} decide.
+... The value `none`.
+** For `coreos.inst.ignition_url=`, specify the URL to the `worker.ign` file. Only HTTP and HTTPS protocols are supported.
+** For `coreos.live.rootfs_url=`, specify the matching rootfs artifact for the kernel and initramfs you are booting. Only HTTP and HTTPS protocols are supported.
+
+** For installations on DASD-type disks, complete the following tasks:
+... For `coreos.inst.install_dev=`, specify `/dev/dasda`.
+... Use `rd.dasd=` to specify the DASD where {op-system} is to be installed.
+... Leave all other parameters unchanged.
++
+Example parameter file, `additional-worker-dasd`, for the bootstrap machine:
```

Comment:
Example parameter file, `additional-worker-dasd`, for the bootstrap machine:
-->
Example parameter file, `additional-worker-dasd.parm`: